### PR TITLE
Sophia - update ride table and add to navbar

### PR DESCRIPTION
**Pull request by SophiaWeiler**
In this PR, we have updated the ride table to display all the backend information + tests + storybook. We also added routes and the navbar for ride request. This finishes all frontend for #7 .

### CODE REVIEW

**Review by mike-k999 — DISMISSED** [review]
I tested this code locally and it fixes some routing issues.  LGTM

File: frontend/src/main/components/Nav/AppNavbar.js [hunk 1]
```diff
@@ -56,6 +56,18 @@ export default function AppNavbar({ currentUser, systemInfo, doLogout, currentUr
                   </NavDropdown>
                 )
               }
+              {
+                (hasRole(currentUser, "ROLE_ADMIN") || hasRole(currentUser, "ROLE_DRIVER") || hasRole(currentUser, "ROLE_USER")) && (
```
Comment (pconrad) [inline frontend/src/main/components/Nav/AppNavbar.js]:
```suggestion
                (hasRole(currentUser, "ROLE_ADMIN") || hasRole(currentUser, "ROLE_DRIVER") || hasRole(currentUser, "ROLE_RIDER")) && (
```

File: frontend/src/main/components/Nav/AppNavbar.js [hunk 2]
```diff
@@ -56,6 +56,18 @@ export default function AppNavbar({ currentUser, systemInfo, doLogout, currentUr
...
+                  <NavDropdown title="Ride Request" id="appnavbar-ride-dropdown" data-testid="appnavbar-ride-dropdown" >
+                    <NavDropdown.Item as={Link} to="/ride/">Rides</NavDropdown.Item>
+                    {
+                      (hasRole(currentUser, "ROLE_USER") || hasRole(currentUser, "ROLE_ADMIN")) && (
```
Comment (pconrad) [inline frontend/src/main/components/Nav/AppNavbar.js]:
```suggestion
                      (hasRole(currentUser, "ROLE_RIDER") || hasRole(currentUser, "ROLE_ADMIN")) && (
```

**Review by zachmiller-ucsb — APPROVED** [review]
lgtm

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**